### ISSUE TITLE
Data race detected in sortyLsw.go

### ISSUE THREAD

**Issue by chkohner** [jfcg/sorty#4]
The data race detector is picking up an issue in long() of sortyLsw.go.

```txt
WARNING: DATA RACE
Write at 0x00c0004f4290 by goroutine 49:
  sync/atomic.AddInt32()
      C:/Program Files/Go/src/runtime/race_amd64.s:305 +0xb
  sync/atomic.AddUint32()
      <autogenerated>:1 +0x1a
  github.com/jfcg/sorty.Sort·dwrap·24()
      C:/Users/chkohner/go/pkg/mod/github.com/jfcg/sorty@v1.2.0/sortyLsw.go:397 +0x64

Previous read at 0x00c0004f4290 by goroutine 47:
  github.com/jfcg/sorty.Sort()
      C:/Users/chkohner/go/pkg/mod/github.com/jfcg/sorty@v1.2.0/sortyLsw.go:406 +0x33c
```

This makes it impossible to write and run unit tests with race detection enabled, because it fails any test that sorty is used in.

**Comment by jfcg** [comment]
Hi @chkohner,

Could you please post:
- your Go and OS versions
- CPU model
- results of the same test on Linux if possible
- results of the same test with latest sorty version
- a minimal example with sorty only that triggers the same race detector failure

Just to make sure, a sorty call must be the only writer on a slice you want to sort.
Thanks..

**Comment by jfcg** [comment]
I was able to recreate the issue with `go test -race` inside latest sorty directory. It is actually known and harmless in that only writes to goroutine counter `ngr` are atomic, but not reads. This is by design for performance reasons, but apparently this causes users real problems when they are utilizing the `-race` flag. I will look into a solution for this soon.

**Comment by jfcg** [comment]
Fixed in `v2.0.4`

**Comment by chkohner** [comment]
Nice. Thanks for the timely fix!
